lib/phoenix_component.ex

```diff
@@ -660,100 +735,112 @@ defmodule Phoenix.Component do
   end
 
   @doc ~S'''
-  Declares attributes for a HEEx function components with compile-time verification and documentation generation.
+  Declares attributes for a HEEx function components.
+
+  ## Arguments
 
-  ## Options
+  * `name` - an atom defining the name of the attribute. Note that attributes cannot define
+  the same name as any other attributes or slots declared for the same component.
+  * `type` - an atom defining the type of the attribute.
+  * `opts` - a keyword list of options. Defaults to `[]`.
 
-    * `:required` - marks an attribute as required. If a caller does not pass
-      the given attribute, a compile warning is issued.
-    * `:default` - the default value for the attribute if not provided
-    * `:doc` - documentation for the attribute
+  ### Types
 
-  ## Types
+  An attribute is declared by its name, type, and options. The following types are supported:
 
-  An attribute is declared by its name, type, and options. The following
-  types are supported:
+  * `:any` - any term.
```

Comment:
we should use the table format here, similar to how Ecto Schema does. 

https://github.com/elixir-ecto/ecto/blob/v3.8.4/lib/ecto/schema.ex#L225